src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php

```diff
@@ -54,6 +54,13 @@ public function __construct(string $host = 'localhost', int $port = 0, bool $tls
         }
         if (!$tls) {
             $stream->disableTls();
+        } elseif (!$verifyPeer) {
+            $stream->setStreamOptions(array_merge_recursive($stream->getStreamOptions(), [
```

Comment:
use `array_replace_recursive` as that's merging associative arrays. Otherwise, things will break when the key already exist in the stream options.
See https://friendsofsymfony.github.io/slides/config_definition.html#6 for the demo (slides are in French, but you care about the code snippet on that slide and the next one, which will be fine).

Reply:
It's better that way ?